cmake/SelectSSH.cmake

```diff
@@ -12,30 +12,32 @@ if(USE_SSH)
 	if(NOT LIBSSH2_FOUND)
 		message(FATAL_ERROR "LIBSSH2 not found. Set CMAKE_PREFIX_PATH if it is installed outside of the default search path.")
 	endif()
-endif()
 
-if(LIBSSH2_FOUND)
-	set(GIT_SSH 1)
-	list(APPEND LIBGIT2_SYSTEM_INCLUDES ${LIBSSH2_INCLUDE_DIRS})
-	list(APPEND LIBGIT2_SYSTEM_LIBS ${LIBSSH2_LIBRARIES})
-	list(APPEND LIBGIT2_PC_LIBS ${LIBSSH2_LDFLAGS})
+	if(LIBSSH2_FOUND)
+		set(GIT_SSH 1)
+		list(APPEND LIBGIT2_SYSTEM_INCLUDES ${LIBSSH2_INCLUDE_DIRS})
+		list(APPEND LIBGIT2_SYSTEM_LIBS ${LIBSSH2_LIBRARIES})
+		list(APPEND LIBGIT2_PC_LIBS ${LIBSSH2_LDFLAGS})
 
-	check_library_exists("${LIBSSH2_LIBRARIES}" libssh2_userauth_publickey_frommemory "${LIBSSH2_LIBRARY_DIRS}" HAVE_LIBSSH2_MEMORY_CREDENTIALS)
-	if(HAVE_LIBSSH2_MEMORY_CREDENTIALS)
-		set(GIT_SSH_MEMORY_CREDENTIALS 1)
+		check_library_exists("${LIBSSH2_LIBRARIES}" libssh2_userauth_publickey_frommemory "${LIBSSH2_LIBRARY_DIRS}" HAVE_LIBSSH2_MEMORY_CREDENTIALS)
+		if(HAVE_LIBSSH2_MEMORY_CREDENTIALS)
+			set(GIT_SSH_MEMORY_CREDENTIALS 1)
+		endif()
+	else()
+		message(STATUS "LIBSSH2 not found. Set CMAKE_PREFIX_PATH if it is installed outside of the default search path.")
```

Comment:
This message seems redundant with the FATAL_ERROR one above. (Won't both be output on the same condition?)

Reply:
One message is about USE_SSH not enabled in cmake, the other is about the package not being found.

I got very confused, because cmake told me it couldn't find libssh2 but it just wasn't enabled as config option (which is now off by default for some reason)

PS: I'm sorry there is now another commit to the PR I wasn't aware it would add to already existing PRs. Fixing it.

Reply:
Oh I see what you mean. I didn't introduce this redundancy, it's already in main.
See: https://github.com/libgit2/libgit2/blob/main/cmake/SelectSSH.cmake
 
What this commit adds is a different message in case of USE_SSH is OFF (which is default now). Otherwise this prints libssh2 not found, just because USE_SSH is off.